cmd/clusterctl/client/tree/discovery.go

```diff
@@ -105,25 +122,98 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	controlPlaneMachines := selectControlPlaneMachines(machinesList)
 	for i := range controlPlaneMachines {
 		cp := controlPlaneMachines[i]
-		addMachineFunc(controlPLane, cp)
+		addMachineFunc(controlPlane, cp)
 	}
 
-	if len(machinesList.Items) == len(controlPlaneMachines) {
-		return tree, nil
+	machinePoolList, err := getMachinePoolsInCluster(ctx, c, cluster.Namespace, cluster.Name)
+	if err != nil {
+		return nil, err
 	}
 
 	workers := VirtualObject(cluster.Namespace, "WorkerGroup", "Workers")
-	tree.Add(cluster, workers)
 
+	if len(machinesList.Items) != len(controlPlaneMachines) {
```

Comment:
I think we should consider a Cluster with both MachineDeployments and Machine pools, so ti should be 

```
if len(machinesList.Items) != len(controlPlaneMachines) || len(machinePoolList.Items) > 0 {
    tree.Add(cluster, workers)
}

if len(machinesList.Items) != len(controlPlaneMachines) {
    ...
}

if len(machinePoolList.Items) > 0 {
    ...
}
```